src/pydap/net.py

```diff
@@ -56,24 +58,32 @@ def create_request(url, session=None):
         # adjust the cookies as needed. We can then use the final url and
         # the final cookies to set up a webob Request object that will
         # be guaranteed to have all the needed credentials:
-        try:
-            # Use session to follow redirects:
-            with closing(session.head(url)) as head:
-                req = Request.blank(head.url)
-
-                # Get cookies from head:
-                cookies_dict = head.cookies.get_dict()
-
-                # Set request cookies to the head cookies:
-                req.headers['Cookie'] = ','.join(name + '=' +
-                                                 cookies_dict[name]
-                                                 for name in cookies_dict)
-                # Set the headers to the session headers:
-                for item in head.request.headers:
-                    req.headers[item] = head.request.headers[item]
-                return req
-        except MissingSchema:
-            # Missing schema can occur in tests when the url
-            # is not pointing to any resource. Simply pass.
-            pass
-    return Request.blank(url)
+        return create_request_from_session(url, session)
+    else:
+        # If a session object was not passed, we simply pass the requests
+        # library. The requests library allows the handling of redirects
+        # that are not naturally handled by Webob.
+        return create_request_from_session(url, requests)
```

Comment:
The ``requests`` module has almost all the same methods as the ``session`` object, so yes, it can be used instead of a ``session`` it our context. I used this approach because it seemed the cleanest but I can see that you disagree :) The alternative would be to replace ``requests`` by ``requests.Session()``, i.e. create a new temporary ``session``.